source/particles/particle_handler.cc

```diff
@@ -1254,11 +1251,11 @@ namespace Particles
                 current_cell->active_cell_index();
 
               particles[active_cell_index].push_back(
-                particles[out_particle->active_cell_index]
+                particles[out_particle->cell->active_cell_index()]
```

Comment:
Couldnt we also do `(*out_particle->particles_on_cell)[out_particle->particle_index_within_cell]` here and below?